src/Texture.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Texture.h"
+
+namespace ccm
+{
+	Texture::Texture(std::string_view textureSource) : m_texture(nullptr, SDL_DestroyTexture)
+	{
+		auto[pTex, width, height, pitch] = TextureGenerator::loadTextureFromFile(textureSource);
+		m_texture.reset(pTex);
+		m_width = width;
+		m_height = height;
+		m_pitch = pitch;
+	}
+
+	Texture::Texture(Texture&& other)
+		:
+		m_texture(std::move(other.m_texture)), m_width(other.m_width),
+		m_height(m_height), m_pitch(m_pitch)
+	{	}
+
+	Texture& Texture::operator=(Texture&& other)
+	{
+		SDL_DestroyTexture(m_texture.get());
```

Comment:
Note: With previous suggestion about `m_texture` deleter, this line is redundant.